.buildkite/commands/should-skip-job.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash -eu
+
+# Usage: should-skip-job.sh --job-type [validation|build|lint]
+# --job-type validation: Skip when changes are limited to documentation, tooling, non-code files, and localization files
+# --job-type build: Skip when changes are limited to documentation, tooling, and non-code files
+# --job-type lint: Skip when changes are limited to documentation, tooling, and non-code files
+
+COMMON_PATTERNS=(
+  "*.md"
+  "*.pot"
+  "*.txt"
+  ".gitignore"
+  "config/**"
+  "docs/**"
+  "fastlane/**"
+  "Gemfile"
+  "Gemfile.lock"
+  "gradle/**"
+  "version.properties"
+)
+
+# Check if arguments are valid
+if [ -z "${1:-}" ] || [ "$1" != "--job-type" ] || [ -z "${2:-}" ]; then
+  echo "Error: Must specify --job-type [validation|build|lint]"
+  buildkite-agent step cancel
+  exit 1
```

Comment:
I wonder if a more differentiating exit code would be beneficial for that (e.g. `127` or `42` or whatever, anything arbitrary but `1`), to differentiate it from the `exit 1` that `pr_changed_files` would use to return an answer of "false"? 🤷 

```suggestion
  exit 15
```
And same for line 40